lib/rubocop/cop/performance/open_struct.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # This cop checks for `OpenStruct.new` calls.
+      # Usually `OpenStruct`s can be easily replaced by static `Structs` or
+      # which generally are much faster.
+
+      # @example
+      #   # bad
+      #   class MyClass
+      #     def my_method
+      #       OpenStruct.new(my_key1: 'my_value1', my_key2: 'my_value2')
+      #     end
+      #   end
+      #
+      #   # good
+      #   class MyClass
+      #     MyStruct = Struct.new(:my_key1, :my_key2)
+      #     def my_method
+      #       MyStruct.new('my_value1', 'my_value2')
+      #     end
+      #   end
+      #
+      class OpenStruct < Cop
+        MSG = 'For performance reasons, consider using `Struct` ' \
```

Comment:
I'd reword this to `Consider using Struct over OpenStruct to optimize the performance`. (or something along those lines).